src/rest/connector/libs/xpresso/implementation.py

```diff
@@ -0,0 +1,381 @@
+
+        >>> from pyats.topology import loader
+        >>> testbed = loader.load('testbed.yaml')
+        >>> device = testbed.devices['xpresso']
+        >>> device.connect(alias='rest', via='rest')
+        >>> device.rest.connected
+        True
+    '''
+    @BaseConnection.locked
+    def connect(self, timeout=30):
+        '''connect to the device via REST
+
+        Arguments
+        ---------
+
+            timeout (int): Timeout value
+
+        Raises
+        ------
+
+        Exception
+        ---------
+
+            If the connection did not go well
+
+        Note
+        ----
+
+        There is no return from this method. If something goes wrong, an
+        exception will be raised.
+
+
+        YAML Example
+        ------------
+        devices:
+            xpresso:
+                os: xpresso
+                connections:
+                    rest:
+                        class: rest.connector.Rest
+                        host : "xpresso.cisco.com"
+                        protocol: http
+                        credentials:
+                            rest:
+                                token: <xpressoaccesstoken>
+
+        Code Example
+        ------------
+
+            >>> from pyats.topology import loader
+            >>> testbed = loader.load('testbed.yaml')
+            >>> device = testbed.devices['xpresso']
+            >>> device.connect(alias='rest', via='rest')
+        '''
+
+        if self.connected:
+            return
+
+
+        if 'host' in self.connection_info:
+            ip = self.connection_info['host']
+        else:
+            ip = self.connection_info['ip'].exploded
+        if 'protocol' in self.connection_info:
+            protocol = self.connection_info['protocol']
+        else:
+            protocol = 'https'
+        if 'port' in self.connection_info:
+            port = self.connection_info['port']
+            self.url = f'{protocol}://{ip}:{port}/'
+        else:
+            self.url = f'{protocol}://{ip}/'
+        self.token = get_token(self)
+
+        self.headers = {
+            'Authorization': '{}'.format(self.token),
+        }
+
+        log.info("Connecting to '{d}' with alias "
+                 "'{a}'".format(d=self.device.name, a=self.alias))
+
+        self.session = requests.Session()
+        self.session.trust_env = False
+        self.session.headers.update(self.headers)
+
+        self._is_connected = True
+        log.info("Connected successfully to '{d}'".format(d=self.device.name))
+
+    @BaseConnection.locked
+    def disconnect(self):
+        '''disconnect the device for this particular alias'''
+
+        log.info("Disconnecting from '{d}' with "
+                 "alias '{a}'".format(d=self.device.name, a=self.alias))
+        try:
+            self.session.close()
+        finally:
+            self._is_connected = False
+        log.info("Disconnected successfully from "
+                 "'{d}'".format(d=self.device.name))
```

Comment:
What exception are you expecting that requires a try block? This will not handle any thrown exception, only hide it. Even if you want to ignore the exception, having an except block will at least allow the opportunity to log the error.

Reply:
may want to fix this for webex implementation too, but I'll fix these